src/providers/postgres/qgspostgresprovider.cpp

```diff
@@ -1126,36 +1126,58 @@ bool QgsPostgresProvider::loadFields()
 
 void QgsPostgresProvider::setEditorWidgets()
 {
-  if ( tableExists( *connectionRO(), EDITOR_WIDGET_STYLES_TABLE ) )
+  if ( ! tableExists( *connectionRO(), EDITOR_WIDGET_STYLES_TABLE ) )
   {
-    for ( int i = 0; i < mAttributeFields.count(); ++i )
+    return;
+  }
+
+  QStringList quotedFnames;
+  foreach ( const QString &name, mAttributeFields.names() )
```

Comment:
```suggestion
  const QStringList fieldNames;
  for ( const QString &name : fieldNames )
```

Reply:
this suggestion isn't clear, partial (fieldNames will be empty?)

Reply:
Sorry, updated.